cmake/checks.cmake

```diff
@@ -1,38 +1,281 @@
-include(CheckIncludeFile)
-include(CheckIncludeFiles)
-include(CheckSymbolExists)
-include(CheckPrototypeDefinition)
-include(CheckFunctionExists)
-include(CheckCXXSourceCompiles)
-include(CheckTypeSize)
-include(CheckLibraryExists)
-include(CheckCXXSourceRuns)
-include(CheckStructHasMember)
-include(CheckCSourceCompiles)
-include(CheckSourceRuns)
-include(TestBigEndian)
-
 macro(magick_check_env)
-  # Check if `closedir' function returns void instead of `int'
+  include(CheckCSourceCompiles)
+  include(CheckCSourceCompiles)
+  include(CheckCSourceRuns)
+  include(CheckFunctionExists)
+  include(CheckIncludeFile)
+  include(CheckIncludeFiles)
+  include(CheckLibraryExists)
+  include(CheckPrototypeDefinition)
+  include(CheckSourceRuns)
+  include(CheckStructHasMember)
+  include(CheckSymbolExists)
+  include(CheckTypeSize)
+  include(TestBigEndian)
+
+  if(BUILD_MAGICKPP)
+    include(CheckCXXSourceCompiles)
+    include(CheckCXXSourceRuns)
+  endif()
+
+  option(MSVC_SKIP_CHECKS "Skip trivial checks on MSVC" ON)
+  if(MSVC AND MSVC_SKIP_CHECKS)
```

Comment:
Just a thought but maybe it would be nicer to make a function to use below that skips on msvc inside so we don't have to update both this list and below when something changes?

Reply:
It's not much of a problem if that list does not get updated, it only results in a check being run. I don't think the checks can be skipped entirely as some of the checks below cannot be predefined, such as x86- vs x86_64-specific flags.